### PR TITLE
SI-6963 Removes -Xmigration switch

### PR DESCRIPTION
**Pull request by JamesIry**
Removes the -Xmigration switch, the internal setting, the documentation,
and the tests around it.

review @gkossakowski

### CODE REVIEW

File: src/compiler/scala/tools/nsc/typechecker/RefChecks.scala [hunk 1]
```diff
@@ -1229,16 +1229,6 @@ abstract class RefChecks extends InfoTransform with scala.reflect.internal.trans
...
     }
 
-    /** Similar to deprecation: check if the symbol is marked with @migration
-     *  indicating it has changed semantics between versions.
-     */
-    private def checkMigration(sym: Symbol, pos: Position) = {
-      if (sym.hasMigrationAnnotation)
-        unit.warning(pos, "%s has changed semantics in version %s:\n%s".format(
-          sym.fullLocationString, sym.migrationVersion.get, sym.migrationMessage.get)
-        )
```
Comment (retronym) [inline src/compiler/scala/tools/nsc/typechecker/RefChecks.scala:1239]: You should also be able to lop `Symbol#{hasMigrationAnnotation, migrationVersion, migrationMessage}` and `Definitions#MigrationAnnotationClass}`.

**Comment by JamesIry** [comment]
@retronym They're used by the docs (see src/compiler/scala/tools/nsc/doc/model/ModelFactory.scala). So unless we're removing the annotation from docs I have to leave those the Symbol methods etc.

**Comment by retronym** [comment]
I would suggest to remove them from the docs (inlining the message into the regular comment if it is still more than marginally relevant.)

**Comment by JamesIry** [comment]
So, just whack the scala.annotation.migration annotation then? @adriaanm, @paulp thoughts?

**Comment by paulp** [comment]
Oh wonderful, I had the foresight to private[scala] it. Yes, 40 whacks with a wet noodle.

**Comment by adriaanm** [comment]
As much as I'd like to slim down 2.11, this may be be more baby than water.
How about we fix -Xmigration to take a version number so it becomes useful again?

**Comment by JamesIry** [comment]
@adriaanm has convinced me. I'm closing this PR and will resubmit with a version that takes an argument.

**Comment by retronym** [comment]
Can we get any data (via straw polls or somesuch) on how many folks used this option in the past? If we keep it on life support, how will we increase that number in the future?

**Pull request closed without merging**